File: M3SDK_Android/M3SDK_Demo/UGR_java/app/src/main/java/net/m3mobile/ugr_demo/LogWriter.java
package net.m3mobile.ugr_demo;

import android.util.Log;

public class LogWriter {

    private static final String TAG = "UGR_java";
    static int logLevel = 0;

    /**
     * Log Level Error
     **/
    public static void e(String message) {
        if (logLevel <= 4)
            Log.e(TAG, buildLogMsg(message));
    }

    /**
     * Log Level Warning
     **/
    public static void w(String message) {
        if (logLevel <= 3)
            Log.w(TAG, buildLogMsg(message));
    }

    /**
     * Log Level Information
     **/
    public static void i(String message) {
        if (logLevel <= 2)
            Log.i(TAG, buildLogMsg(message));
    }

    /**
     * Log Level Debug
     **/
    public static void d(String message) {
        if (logLevel <= 1)
            Log.d(TAG, buildLogMsg(message));
    }

    /**
     * Log Level Verbose
     **/
    public static void v(String message) {
        if (logLevel <= 0)
            Log.v(TAG, buildLogMsg(message));
    }

    private static String buildLogMsg(String message) {
        StackTraceElement stackTraceElement = Thread.currentThread().getStackTrace()[4];
        return "[" + stackTraceElement.getFileName().replace(".java", "") + "::" + stackTraceElement.getMethodName() + "]" + message;
    }
}

File: M3SDK_Android/M3SDK_Demo/UGR_kotlin/app/src/main/java/net/m3mobile/ugr_demo/LogWriter.kt
package net.m3mobile.ugr_demo

import android.util.Log

object LogWriter {
    private const val TAG = "UGR_kotlin"
    var logLevel = 0

    /**
     * Log Level Error
     */
    fun e(message: String) {
        if (logLevel <= 4) Log.e(TAG, buildLogMsg(message))
    }

    /**
     * Log Level Warning
     */
    fun w(message: String) {
        if (logLevel <= 3) Log.w(TAG, buildLogMsg(message))
    }

    /**
     * Log Level Information
     */
    fun i(message: String) {
        if (logLevel <= 2) Log.i(TAG, buildLogMsg(message))
    }

    /**
     * Log Level Debug
     */
    fun d(message: String) {
        if (logLevel <= 1) Log.d(TAG, buildLogMsg(message))
    }

    /**
     * Log Level Verbose
     */
    fun v(message: String) {
        if (logLevel <= 0) Log.v(TAG, buildLogMsg(message))
    }

    private fun buildLogMsg(message: String): String {
        val stackTraceElement = Thread.currentThread().stackTrace[4]
        return "[" + stackTraceElement.fileName.replace(
            ".java",
            ""
        ) + "::" + stackTraceElement.methodName + "]" + message
    }
}